File: DSA/Stack/C/InfixToPostfix.c
#include<stdio.h>
#include<ctype.h>
char stack[100];
int top=-1;
char pop(){
    if(top==-1){
        return -1;
    }
    else{
        return stack[top--];
    }

}
void push(char e){
    top=top+1;
    stack[top]=e;
}
int priority(char e){
    if(e=='('){
        return 0;
    }
    if(e=='+'||e=='-'){
        return 1;
    }
    if(e=='*'||e=='/'){
        return 2;
    }
    if(e=='^')
        return 3;
    return 0;
}
int main(){
    char str[100];
    char *e,x;
    printf("Enter the line: ");
    scanf("%s",str);
    printf("\n");
    e=str;
    while(*e!='\0'){
        if(isalnum(*e)){
            printf("%c",*e);
        }
        else if(*e=='('){
            push(*e);
        }
        else if(*e == ')')
        {
            while((x = pop()) != '(')
                printf("%c", x);
        }
        else{
            while(priority(stack[top])>=priority(*e)){
                printf("%c",pop());
            }
            push(*e);
        }
        e++;
    }
    while(top!=-1){
        printf("%c",pop());
    }
    return 0;
}

File: DSA/Stack/python/InfixToPostfix.py
#Infix To Postfix Conversion Only

class Stack:
    def __init__(self):
        self.data = []

    def push(self, data):
        self.data.append(data)

    def pop(self) -> chr:
        if self.data:
            return self.data.pop()

    def peek(self) -> chr:
        if self.data:
            return self.data[-1]


def priority(item) -> int:
    if item == '(':
        return 0
    elif item in '+-':
        return 1
    elif item in '*/':
        return 2
    elif item == '^':
        return 3
    return 0


def infix_to_postfix(infix_expr: str) -> str:
    stack = Stack()
    postfix_expr = ''
    for c in infix_expr:
        if c.isalnum():
            postfix_expr += c
        elif c == '(':
            stack.push(c)
        elif c in "+-*/^":
            if not stack.data:
                stack.push(c)
            else:
                while stack.data and priority(stack.peek()) >= priority(c):
                    postfix_expr += stack.pop()
                stack.push(c)
        elif c == ')':
            while stack.data and stack.peek() != '(':
                postfix_expr += stack.pop()
            stack.pop()

        # print(c.center(len(infix_expr)), end='|')
        # print(f'{"".join(stack.data)}'.center(len(infix_expr)), end='|')
        # print(f'{postfix_expr}'.center(len(infix_expr)))

    # print("stack:", stack.data)
    while stack.data:
        postfix_expr += stack.pop()

    return postfix_expr

if __name__=="__main__":
    infix_expr = input("Enter the Expression: ")
    postfix_expr = infix_to_postfix(infix_expr)
    print(postfix_expr)